### PR TITLE
Adds Medieval Pirates (and small assault_pod change)

### PR DESCRIPTION
**Pull request by uaioy**
## About The Pull Request

(short, slightly outdated video compilation of the pirates) https://youtu.be/j9BNFzcUSjc

Adds the **MEDIEVAL WARMONGERS** pirate gang, these pirates are focused on causing chaos directly on the station with the use of melee weaponry and swarming tactics.
they are not very smart when it comes to using technology, so they crash in the station in an attempt to dock.

the pirates spawn with the NOGUN trait as they have never used guns before but have access to dashing and the unbreakable component(small heal burst upon reaching crit).
Warlord spawns with a better version of hulk(isnt removed upon crit and can still do stuff while critted) and gigantism mutation by default because ~~Chad Maximus~~ the other pirates require someone to run to when stuff like flashbangs, batons or lasers are present. The warlord spawns with a boarding axe which is a better axe that is good at destroying things and a meat hook. **there's a total of 5 pirates.**

the pirate shuttle has:
- thermite barrel and 3 large beakers
- single weak claymore
- 3 extra military spears
- 3 kite shields
- 3 heavy bolas
- 4 regular bolas
- a stack of wood
- 2 stacks of mourning poultice (15 uses each)
- one burn medkit

**shuttle was made by striders18, big thanks to him**
image

things added:

- the new pirate gang
- a gamberson jumpsuit
- crude armor (good melee and wound protection but lacks on the rest)
- warlord armor (extreme protection at the cost of slowdown)
- a shortsword(bootleg claymore with 30 block chance and goes on belt instead of back)
- military spear (better at throwing, better wounding, deals blunt wounds instead of sharp)
- axe with better damage and demolition mod for the warlord
- a var to specify the name of the landing zone destination for the assaultpod
- an assault pod targeting device item for the medieval shuttle
- a thermite barrel
- kite shields
- new hulk variant that doesnt go away when crit and has tenacity
## Why It's Good For The Game
I always liked the chaotic nature of the russian bounty hunters that would come from their extra numbers and lack of tools. i thought something of a similar idea on pirates would be good, a very direct and scary militia rushing in with sword and faith would create some awesome scenarios

## Changelog
:cl:
add: New heavy weight pirates, medieval warmongers
add: Adds military spear, shortsword, boarding axe, kite shields
add: Adds warlord and crude armor
add: Adds medieval shuttle (made by striders18)
/:cl:

### CODE REVIEW

**Comment by uaioy** [comment]
agh i just need to fix why the sprites arent working

**Comment by another-expler** [comment]
Ahoy, i like them! But i think they need some kind of ranged weapon, like bow or crossbow, maybe flintlock or musket, because of slowdown they will be killed with egun in a minute

**Comment by Singul0** [comment]
> they will be killed with egun in a minute

The laser buff and it's consequences has been a disaster to the human race

**Comment by DaCoolBoss** [comment]
I'm also concerned about these guys being competitive. Skeleton and lizard pirates have similar melee capability but have additional options (good ones at that) to cover melee's weaknesses. I don't think giving them ranged crossbows or whatever is the way to go because that dilutes their gimmick, but maybe some kind of ability to close gaps like ling adrenals or lobster organs?

**Comment by another-expler** [comment]
> I'm also concerned about these guys being competitive. Skeleton and lizard pirates have similar melee capability but have additional options (good ones at that) to cover melee's weaknesses. I don't think giving them ranged crossbows or whatever is the way to go because that dilutes their gimmick, but maybe some kind of ability to close gaps like ling adrenals or lobster organs?

Maybe something throwing like bolas, Spears,  jawelins, or knives... Maybe batches?

**Comment by uaioy** [comment]
> I'm also concerned about these guys being competitive. Skeleton and lizard pirates have similar melee capability but have additional options (good ones at that) to cover melee's weaknesses. I don't think giving them ranged crossbows or whatever is the way to go because that dilutes their gimmick, but maybe some kind of ability to close gaps like ling adrenals or lobster organs?

well idk, i originally wanted to give them bow but they count as guns so they couldnt use it so i just gave them more spears, the idea is that warlord can soak up some laser hits and the other 4 pirates can swarm the gunner, maybe with some casualties

File: code/modules/shuttle/assault_pod.dm [hunk 1]
```diff
@@ -61,3 +62,22 @@
...
+	desc = "A map of the station used to select where you want to land your shuttle."
+	shuttle_id = "pirate"
+	width = 8
+	lz_dir = 8
+	lzname = "pirate"
+
+/obj/item/assault_pod/medieval/Initialize(mapload)
+	. = ..()
+	var counter = length(SSmachines.get_machines_by_type_and_subtypes(/obj/machinery/computer/shuttle/pirate))
+	counter++
```
Comment (MrMelbert) [inline code/modules/shuttle/assault_pod.dm:80]:
```suggestion
	var/counter = 1 + length(SSmachines.get_machines_by_type_and_subtypes(/obj/machinery/computer/shuttle/pirate))
```

File: code/modules/clothing/suits/armor.dm [hunk 2]
```diff
@@ -644,3 +644,47 @@
...
+	icon_state = "military"
+	inhand_icon_state = "armor"
+	dog_fashion = null
+	armor_type = /datum/armor/military
+	allowed = list(
+		/obj/item/banner,
+		/obj/item/claymore/shortsword,
+		/obj/item/nullrod,
+		/obj/item/spear,
+		)
```
Comment (MrMelbert) [inline code/modules/clothing/suits/armor.dm:660]:
```suggestion
	)
```

File: code/modules/antagonists/pirate/pirate_gangs.dm [hunk 3]
```diff
@@ -205,3 +205,25 @@ GLOBAL_LIST_INIT(heavy_pirate_gangs, init_pirate_gangs(is_heavy = TRUE))
...
+	name = "Medieval Warmongers"
+
+	is_heavy_threat = TRUE
+	ship_template_id = "medieval"
+	ship_name_pool = "medieval_names"
+
+	threat_title = "HOMAGE PAYMENT REQUEST"
+	threat_content = "SALUTATIONS, THIS IS %SHIPNAME AND WE ARE COLLECTING MONEY \
+		FROM THE VASSALS IN OUR TERRITORY, YOU JUST SO HAPPEN TO BE IN IT TOO!! NORMALLY \
+		WE WOULD SLAUGHTER ALL OF YOU AND PILLAGE YOUR BELONGINGS, BUT WE ARE WILLING \
```
Comment (MrMelbert) [inline code/modules/antagonists/pirate/pirate_gangs.dm:220]: Why would they slaughter and pillage their own vassals? This is a Feudalism Fail

Comment (uaioy) [reply code/modules/antagonists/pirate/pirate_gangs.dm:220]: i wanted to mean it as if you were "intruding in their territory" and thats why they had reasons to steal your shit

File: code/modules/clothing/suits/armor.dm [hunk 4]
```diff
@@ -644,3 +644,47 @@
...
+	laser = 20
+	energy = 25
+	bomb = 25
+	fire = 10
+	acid = 50
+	wound = 20
+
+/obj/item/clothing/suit/armor/riot/knight/warlord
+	name = "golden plate armor"
+	desc = "This bulky set of armor coated with a shiny layer of gold. it seems to almost reflect all light sources."
```
Comment (EEASAS) [inline code/modules/clothing/suits/armor.dm:674]: "This bulky set of armor is coated with a shiny layer of gold. It seems to almost reflect all light sources"

**Comment by EEASAS** [comment]
there's a lot of descriptions with sub-par grammar like missing capitalization

**Comment by uaioy** [comment]
> there's a lot of descriptions with sub-par grammar like missing capitalization

can you specify where so i can fix it?

**Comment by EEASAS** [comment]
> > there's a lot of descriptions with sub-par grammar like missing capitalization
> 
> can you specify where so i can fix it?

swordsman's under and shortsword

**Comment by uaioy** [comment]
> swordsman's under and shortsword the sleeping bag's flavour text and description also
ok i fixed the missing capitalization, is it ok now?

**Comment by EEASAS** [comment]
> > swordsman's under and shortsword the sleeping bag's flavour text and description also
> 
> ok i fixed the missing capitalization, is it ok now?

seems good now

**Comment by Loafin34** [comment]
Good to see this PR, man! Are the pirates able to store enough spears to attack their opponents at range effectively? I fear they might run out of 'ammo' easily and be forced to run head-first into a bunch of laser guns.

**Comment by uaioy** [comment]
> Good to see this PR, man! Are the pirates able to store enough spears to attack their opponents at range effectively? I fear they might run out of 'ammo' easily and be forced to run head-first into a bunch of laser guns.

at the moment they can only carry a spear on their suit and their back, by default they spawn with a spear on their suit and can grab more from the ship.
do you think it would be alright to make them spawn with 2 spears by default?

**Comment by Loafin34** [comment]
it's fine with them having 2 spears. around how much is the fracture chance with the spears? that would help the pirates a lot in closing the distance between their foes.

**Comment by uaioy** [comment]
> it's fine with them having 2 spears. around how much is the fracture chance with the spears? that would help the pirates a lot in closing the distance between their foes.

throwforce = 25
wound_bonus = 0 (the regular spear is -15)
bare_wound_bonus = 20
throw_range = 9
throw_speed = 5

**Comment by carlarctg** [comment]
You should add a crossbowman/musketeer/whatever so they have some versatility. To get around the NOGUNS issue, give this one chunky fingers instead, and make sure their ranged weapon doesn't get blocked by the trait. Perhaps link the trait to some bulky, unremovable gloves to make it clear.

**Comment by uaioy** [comment]
> You should add a crossbowman/musketeer/whatever so they have some versatility. To get around the NOGUNS issue, give this one chunky fingers instead, and make sure their ranged weapon doesn't get blocked by the trait. Perhaps link the trait to some bulky, unremovable gloves to make it clear.

ill do something similar as a future PR to buff the pirates in case they turn out to be underwhelming for the crew, but rn i wanna see the general reception and how they fare in the game before doing anything else

**Comment by uaioy** [comment]
> Very nice, one problem though, the boarding axe looks way too crude and makeshift, I would make it more of a war axe and make the blade look metallic and sharpened

that was kind of the intention to potray that its a thing that OBLITERARTES structures and to make it look more barbaric since well, they are space medieval warmongers

**Comment by relativelysmalllebowski** [comment]
boy oh boy cant wait to see these pirates get peppersprayed and batoned as soon as they deploy

**Comment by uaioy** [comment]
> Yeah I just realised they can be batonned once and die, you should give them some stun resistance probably since they are like the masters of melee combat. Dying to a assistant with a makeshift stunprod while you're a living skeleton that spent years practicing the blade doesn't sound nice, especially when these pirates can't use most ranged weapons (if any really) in comparison to regular pirates that indeed can.
> 
> Now that I think about it, lack of stun resistance just makes them a worse version of the default pirates if they can't use laser guns

there's already 5 of em and one of them is a literal hulk tho, thatd be kinda kinda overkill
also yeah they are in the chameleon kit since it lets you change it to everything it seems

**Comment by uaioy** [comment]
thinking of replacing the extra spears on shuttle with a single claymore

**Comment by jlsnow301** [comment]
cool concept. played nice in a few rounds. just check up on future balance issues if they exist

**Pull request merged**